pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/Node.java

```diff
@@ -224,11 +224,25 @@
      * @param xpathString
      *            the expression to check
      * @return List of all matching nodes. Returns an empty list if none found.
-     * @throws JaxenException
+     * @throws JaxenException if the xpath is incorrect or fails altogether
      */
     List<? extends Node> findChildNodesWithXPath(String xpathString) throws JaxenException;
 
     /**
+     * Returns all the nodes matching the xpath expression and casts the nodes to a specific type.
+     *
+     * @param clazz
+     *            the class of the result component type
+     * @param xpathString
+     *            the expression to check
+     * @return List of all matching nodes. Returns an empty list if none found.
+     *
+     * @throws JaxenException if the xpath is incorrect or fails altogether
+     * @throws ClassCastException if the nodes found with the xpath exception cannot be cast.
+     */
+    <T extends Node> List<T> findChildNodesWithXPathChecked(Class<T> clazz, String xpathString) throws JaxenException;
```

Comment:
Right I can remove it. We could still revert the commit anyway (btw a place such as [`ClassTypeResolverTest`](https://github.com/pmd/pmd/blob/master/pmd-java/src/test/java/net/sourceforge/pmd/typeresolution/ClassTypeResolverTest.java#L252) could use it, though as long as we don't have duplicated code I agree it's best to keep it that way).